### PR TITLE
chore(sdk): v2 compiler compiles to pipeline job spec

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
- Fix capitalization in the proto file.
- compiles to pipeline job spec.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: sdk/python/kfp/v2/compiler/compiler.py [hunk 1]
```diff
@@ -172,11 +172,36 @@ def _create_pipeline(
 
     return pipeline_spec
 
+  def _create_pipeline_job(
```
Comment (NikeNano) [inline sdk/python/kfp/v2/compiler/compiler.py:175]: Really nice typing :)

File: sdk/python/kfp/v2/compiler/compiler.py [hunk 2]
```diff
@@ -172,11 +172,36 @@ def _create_pipeline(
...
+      self,
+      pipeline_spec: pipeline_spec_pb2.PipelineSpec,
+      pipeline_root: Optional[str] = None,
```
Comment (numerology) [inline sdk/python/kfp/v2/compiler/compiler.py:178]: IIUC pipeline_root is still required for now?

File: sdk/python/kfp/v2/compiler/compiler_utils.py [hunk 3]
```diff
@@ -52,3 +53,40 @@ def to_message(param: dsl.PipelineParam):
     return result
 
   return {param.name: to_message(param) for param in pipeline_params}
+
+
+def build_runtime_config_spec(
+    pipeline_root: Optional[str] = None,
```
Comment (numerology) [inline sdk/python/kfp/v2/compiler/compiler_utils.py:59]: Ditto. I guess for now pipeline root is still required, right?

File: sdk/python/kfp/v2/compiler/compiler_utils.py [hunk 4]
```diff
@@ -52,3 +53,40 @@ def to_message(param: dsl.PipelineParam):
...
+  Args:
+    pipeline_root: The root of pipeline outputs. Optional.
+    pipeline_parameters: The mapping from parameter names to values. Optional.
+
+  Returns:
+    A pipeline job RuntimeConfig object.
+  """
+
+  def _get_value(
+      value: Union[int, float, str]) -> Optional[pipeline_spec_pb2.Value]:
```
Comment (numerology) [inline sdk/python/kfp/v2/compiler/compiler_utils.py:73]: Given `None` is allowed in the function, shall we do `Optional[Union[int, float, str]]` here?

File: sdk/python/kfp/v2/compiler/compiler.py (hunk 1 above)
Comment (chensun) [reply sdk/python/kfp/v2/compiler/compiler.py:175]: Thanks!

File: sdk/python/kfp/v2/compiler/compiler.py [hunk 5]
```diff
@@ -172,11 +172,36 @@ def _create_pipeline(
...
+
   def compile(self,
               pipeline_func: Callable[..., Any],
               output_path: str,
               type_check: bool = True,
-              pipeline_name: str = None) -> None:
+              pipeline_name: Optional[str] = None,
+              pipeline_root: Optional[str] = None,
+              pipeline_parameters: Optional[Mapping[str, Any]] = None) -> None:
     """Compile the given pipeline function into workflow yaml.
```
[the remark this answers is not in the document]
Comment (chensun) [reply sdk/python/kfp/v2/compiler/compiler.py]: Good catch!

File: sdk/python/kfp/v2/compiler/compiler.py (hunk 2 above)
Comment (chensun) [reply sdk/python/kfp/v2/compiler/compiler.py]: Updated.

File: sdk/python/kfp/v2/compiler/compiler_utils.py (hunk 4 above)
Comment (chensun) [reply sdk/python/kfp/v2/compiler/compiler_utils.py]: Good catch!

File: sdk/python/kfp/v2/compiler/compiler.py [hunk 6]
```diff
@@ -172,41 +172,72 @@ def _create_pipeline(
...
+    """Compile the given pipeline function into pipeline job json.
 
     Args:
       pipeline_func: Pipeline function with @dsl.pipeline decorator.
+      pipeline_root: The root of the pipeline ouputs.
       output_path: The output pipeline spec .json file path. for example,
         "~/a.json"
       type_check: Whether to enable the type check or not, default: True.
       pipeline_name: The name of the pipeline. Optional.
+      pipeline_parameters: The mapping from parameter names to values. Optional.
```
Comment (numerology) [inline sdk/python/kfp/v2/compiler/compiler.py:214]: optional nit: make the order of docstring consistent with function signature

File: sdk/python/kfp/v2/compiler/compiler_test.py [hunk 7]
```diff
@@ -65,7 +65,10 @@ def simple_pipeline(pipeline_input='Hello KFP!',):
             input_value=producer.outputs['output_value'])
 
       target_json_file = os.path.join(tmpdir, 'result.json')
-      compiler.Compiler().compile(simple_pipeline, target_json_file)
+      compiler.Compiler().compile(
+          pipeline_func=simple_pipeline,
+          pipeline_root='duumy_root',
```
Comment (numerology) [inline sdk/python/kfp/v2/compiler/compiler_test.py:70]:
```suggestion
          pipeline_root='dummy_root',
```

**Review by numerology — COMMENTED** [review]
Thanks! Mostly LGTM modulo two nit comments

File: sdk/python/kfp/v2/compiler/compiler.py (hunk 6 above)
Comment (chensun) [reply sdk/python/kfp/v2/compiler/compiler.py:213]: Fixed.

File: sdk/python/kfp/v2/compiler/compiler_test.py (hunk 7 above)
Comment (chensun) [reply sdk/python/kfp/v2/compiler/compiler_test.py]: Good catch!

**Pull request merged**